src/markdown-pages/build-apps/build-visualization.mdx

```diff
@@ -0,0 +1,210 @@
+  - title: New Relic VSCode extension pack
+    url: https://marketplace.visualstudio.com/items?itemName=new-relic.new-relic-extension-pack
+tags:
+  - nr1 cli
+  - NR One Catalog
+  - Subscribe visualizations
+---
+
+<Intro>
+
+Building your own visualizations for New Relic dashboards enables you and your organization to query data from any source and display it on your dashboard. In this guide, you'll learn how to use the CLI to generate a visualization file structure, run the visualization locally in New Relic One's Visualization Explorer where you can quickly test and iterate, and deploy the visualization for you and others to use on dashboards.
+
+</Intro>
+
+## Before you begin
+
+To get started, make sure you have a [New Relic](https://newrelic.com/signup/) account.
+
+To develop projects, you need the New Relic One CLI (command line interface). If you haven't already installed it, do the following:
+
+- Install [Node.js](https://nodejs.org/en/download/).
+- Complete all the steps in the [CLI quick start](https://one.newrelic.com/launcher/developer-center.launcher?pane=eyJuZXJkbGV0SWQiOiJkZXZlbG9wZXItY2VudGVyLmRldmVsb3Blci1jZW50ZXIifQ==).
+
+## Create a new visualization
+
+In this first set of steps you use the CLI to quickly generate the boilerplate for a new visualization.
+
+<Steps>
+
+  <Step>
+
+Ensure you're working with the latest version of the New Relic One CLI:
+
+```bash
+nr1 update
+```
+
+  </Step>
+
+  <Step>
+
+Start with a new Nerdpack. If you just installed the New Relic One CLI, then you have a new Nerdpack project directory (called something like, `my-awesome-nerdpack`). If you need to create one, run `nr1 create` and select the `nerdpack` option.
+
+```bash
+nr1 create --type nerdpack --name my-awesome-nerdpack
+```
+
+  </Step>
+
+  <Step>
+
+Navigate to the root of your Nerdpack project, and then run the `create` command.
+
+```bash
+cd my-awesome-nerdpack
+nr1 create --type visualization --name my-awesome-visualization
+```
+
+As a result, you have a new visualization directory matching the name you gave your visualization. It is located in the `visualizations` directory in the root of your nerdpack project.
+
+<img
+  src="../../images/build-an-app/visualization-boilerplate-files.png"
+  alt="Visualization files generated by CLI"
+/>
+
+The files created provide an example visualization – a radar chart populated by a basic NRQL query.
+
+The `nr1.json` file provides metadata including the `configuration` field that defines the prop input fields to be shown in the UI. These are the fields users will fill in to create an instance of the visualization. Supported input types are `boolean`, `string`, `number`, `json`, `nrql`, `namespace`, and `collection`. A `namespace` type groups a set of props together in the prop editing UI under a heading. A `collection` defines a set of fields that can be repeated and result in an array prop type. The `nrqlQueries` entry is an example of a collection type.
```

Comment:
This may be better presented in a our table component so you can define each one of these more visually.  https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/README.md#table